drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -290,11 +288,11 @@ static int flash_npcx_nor_erase(const struct device *dev, off_t addr, size_t siz
 	while (size > 0) {
 		flash_npcx_uma_cmd_only(dev, SPI_NOR_CMD_WREN);
 		/* Send page/block erase command with addr */
-		if ((size >= BLOCK_64K_SIZE) && offset_64k_aligned) {
+		if ((size >= BLOCK_64K_SIZE) && SPI_NOR_IS_64K_ALIGNED(addr)) {
 			flash_npcx_uma_cmd_by_addr(dev, SPI_NOR_CMD_BE, addr);
 			addr += BLOCK_64K_SIZE;
 			size -= BLOCK_64K_SIZE;
-		} else if ((size >= BLOCK_4K_SIZE) && offset_4k_aligned) {
+		} else if ((size >= BLOCK_4K_SIZE) && SPI_NOR_IS_SECTOR_ALIGNED(addr)) {
```

Comment:
Actually you can just have `else` here and drop the `else` that just returns `-EINVAL`, because:
 - in line 269 you ensure that `addr` is at least sector aligned, assuming that every 64k block is also sector aligned, this means that you have either 64k block or sector erases; block is covered by 291 so sector is everything else;
 - in line 277 size is checked to be 4k aligned, which makes it automatically 64k block aligned, so anything that reaches the loop, as a size, is at least 4k aligned; because line 291 covers the block alignment then everything else is just 4k
 - the `-EINVAL` will never be reported from loop because this kind of `size` error would be caught by line 275, before even loop is reached.
 
You will save some code-flash and clocks (unless you hit some strange case).

Reply:
I included suggested changes in latest version of PR. Thank you